zipline/pipeline/factors/factor.py

```diff
@@ -626,7 +622,9 @@ def rank(self, method='ordinal', ascending=True, mask=NotSpecified):
         return Rank(self, method=method, ascending=ascending, mask=mask)
 
     @expect_types(
-        target=Slice, correlation_length=int, mask=(Filter, NotSpecifiedType),
+        target=ComputableTerm,
```

Comment:
I would probably put `(Factor, Slice)` here rather than `ComputableTerm`.  We shouldn't accept a Filter or a Classifier here.